webapp/src/errorList.tsx

```diff
@@ -133,17 +107,70 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
         })
     }
 
-    private generateStackTraces(exception: pxsim.DebuggerBreakpointMessage) {
+    getCompilerErrors(errors: pxtc.KsDiagnostic[]) {
+        function errorKey(error: pxtc.KsDiagnostic): string {
+            // React likes have a "key" for each element so that it can smartly only
+            // re-render what changes. Think of it like a hashcode/
+            return `${error.messageText}-${error.fileName}-${error.line}-${error.column}`
+        }
+
         return <div className="ui selection list">
-            {(exception.stackframes || []).map((sf, index) => {
-                const location = this.state.callLocations[sf.callLocationId];
+            {(errors).map((e, index) => <ErrorListItem key={errorKey(e)} index={index} error={e} revealError={this.onErrorMessageClick} />)}
+        </div>
+    }
 
-                if (!location) return null;
+    generateStackTraces(exception: pxsim.DebuggerBreakpointMessage) {
+        return <div>
+            <div className="exceptionMessage">{pxt.Util.rlf(exception.exceptionMessage)}</div>
+            <div className="ui selection list">
+                {(exception.stackframes || []).map((sf, index) => {
+                    const location = this.state.callLocations[sf.callLocationId];
 
-                return <div className="item stackframe" key={index}>
-                    {lf("at {0} ({1}:{2}:{3})", sf.funcInfo.functionName, sf.funcInfo.fileName, location.line + 1, location.column + 1)}
-                </div>
-            })}
+                    if (!location) return null;
+
+                    return <ErrorListItem key={index} index={index} stackFrame={sf} location={location} revealError={this.onErrorMessageClick}/>
+                })}
+            </div>
         </div>;
     }
 }
+
+interface ErrorListItemProps {
+    index: number;
+    error?: pxtc.KsDiagnostic;
+    stackFrame?: pxsim.StackFrameInfo;
+    location?: pxtc.LocationInfo;
+    revealError: (location: pxtc.LocationInfo, index: number) => void;
+}
+
+interface ErrorListItemState {
+}
+
+
+class ErrorListItem extends React.Component<ErrorListItemProps, ErrorListItemState> {
+    constructor(props: ErrorListItemProps) {
+        super(props)
+
+        this.onErrorListItemClick = this.onErrorListItemClick.bind(this)
+    }
+
+    render() {
+        const {error, stackFrame, location} = this.props
+
+        const message = stackFrame ?  lf("line {0} at {1}", location.line + 1, stackFrame.funcInfo.functionName)
+            : lf("Line {0}: {1}", error.endLine ? error.endLine + 1 : error.line + 1, error.messageText)
+
+        return <div className={`item ${stackFrame ? 'stackframe' : ''}`} role="button"
+                    onClick={this.onErrorListItemClick}
+                    onKeyDown={sui.fireClickOnEnter}
+                    aria-label={lf("Go to {0}", stackFrame ? 'stackframe' : 'error')}
```

Comment:
Maybe include the message in the aria label - so same as current w/ message at end (aria label usually replaces message https://www.w3.org/TR/WCAG20-TECHS/ARIA8.html#ARIA8-ex1)

also nitpick: saying stackframe feels off as you're going to the line the stackframe originated from / not viewing a stack frame that currently exists like you would in the debugger. Maybe worth splitting into two messages / translations?

```suggestion
                    aria-label={lf("Go to {0}: {1}", stackFrame ? '' : 'error', message)}
```